test/Microsoft.ML.Core.Tests/UnitTests/TestModelLoad.cs

```diff
@@ -0,0 +1,34 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.ML.Runtime.Data;
+using Microsoft.ML.Runtime.Model;
+using Microsoft.ML.TestFramework;
+using System.IO;
+using Xunit;
+
+namespace Microsoft.ML.Runtime.RunTests
+{
+    public class TestModelLoad
+    {
+        /// <summary>
+        /// Tests loading a model file that was saved using an older version still loads correctly.
+        /// </summary>
+        [Fact]
+        public void LoadOriginalBinaryLoaderModel()
+        {
+            using (var env = new LocalEnvironment()
+                .AddStandardComponents())
+            using (var modelStream = File.OpenRead("OriginalVersionBinaryLoader.zip"))
```

Comment:
>OriginalVersionBinaryLoader [](start = 52, length = 27)

Do we have this file in repo?

Reply:
It comes through the TestModels NuGet package I added.

Reply:
Is this sustainable? Suppose Scott wants to add a couple more models to test back-compat. How can he do this?

---
In reply to: [226122081](https://github.com/dotnet/machinelearning/pull/1290#discussion_r226122081) [](ancestors = 226122081)

Reply:
Yes, it is sustainable. I am following the same pattern we use for ONNX and TensorFlow test models.

You check the model file into a folder in https://github.com/dotnet/machinelearning-testdata. For example, if Scott has more ML.NET model files, he adds them under https://github.com/dotnet/machinelearning-testdata/tree/master/Microsoft.ML.TestModels, sends a PR and merges. Then someone builds the NuGet package from that repo, and uploads it to our NuGet feed: https://dotnet.myget.org/feed/dotnet-core/package/nuget/Microsoft.ML.TestModels.

The NuGet packages are versioned, so as new versions are created (with new or updated model files), we can just bump the version number in this repo, and start using the new models in the test code.